homeassistant/components/fritzbox/climate.py

```diff
@@ -176,17 +180,17 @@ async def async_set_preset_mode(self, preset_mode):
             await self.async_set_temperature(temperature=self.device.eco_temperature)
 
     @property
-    def min_temp(self):
+    def min_temp(self) -> int:
         """Return the minimum temperature."""
         return MIN_TEMPERATURE
 
     @property
-    def max_temp(self):
+    def max_temp(self) -> int:
         """Return the maximum temperature."""
         return MAX_TEMPERATURE
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str | int | bool | None]:
```

Comment:
It's probably simpler to define proper `TypedDict` when you have this.

Reply:
I don't think HA specify list of attributes which can be used here. Therefore it can't be defined in core and parent class should be typed as `Mapping[str, Any]`. Here you can define your own `TypedDict` just for this component.